### PR TITLE
Revert ConfigurationException due to binary incompatibility

### LINKED ISSUE #6200: #6167 is not backward compatible
**Version Information**
Version of Akka.NET? 1.4.44
Which Akka.NET Modules? Core

**Describe the bug**
A user reported that #6167 causes Lighthouse to crash because it could not find `public static ConfigurationException NullOrEmptyConfig<T>(string path = null)` method fingerprint in `ConfigurationException` class

### PR DESCRIPTION
**Pull request by Arkatufus**
Fixes #6200

## Changes
Revert changes to the static method

### CODE REVIEW

File: src/core/Akka/Configuration/ConfigurationException.cs [hunk 1]
```diff
@@ -16,11 +16,11 @@ namespace Akka.Configuration
     /// </summary>
     public class ConfigurationException : AkkaException
     {
-        public static ConfigurationException NullOrEmptyConfig<T>(string path = null, string reason = null)
+        public static ConfigurationException NullOrEmptyConfig<T>(string path = null)
```
Comment (Arkatufus) [inline src/core/Akka/Configuration/ConfigurationException.cs:19]: Reverting this back to the old method fingerprint

File: src/core/Akka.Cluster/ClusterSettings.cs [hunk 2]
```diff
@@ -35,7 +35,7 @@ public ClusterSettings(Config config, string systemName)
             //TODO: Requiring!
             var clusterConfig = config.GetConfig("akka.cluster");
             if (clusterConfig?.GetConfig("failure-detector") == null)
-                throw ConfigurationException.NullOrEmptyConfig<ClusterSettings>("akka.cluster", "Did you forgot to set the 'akka.cluster.provider' HOCON property to 'cluster'?");
+                throw new ConfigurationException($"Failed to instantiate {nameof(ClusterSettings)}: Configuration does not contain `akka.cluster` node. Did you forgot to set the 'akka.cluster.provider' HOCON property to 'cluster'?");
```
Comment (Arkatufus) [inline src/core/Akka.Cluster/ClusterSettings.cs:38]: Just bake the exception message instead of using the helper method

**Review by Aaronontheweb — CHANGES_REQUESTED** [review]
Needs API approvals

**Comment by Arkatufus** [comment]
Done

**Review by Aaronontheweb — APPROVED** [review]

**Comment by Aaronontheweb** [comment]
@Arkatufus going to give tests a chance to run, then we can push a new release

**Pull request merged**